python/cuml/preprocessing/model_selection.py

```diff
@@ -24,6 +24,109 @@
 from typing import Union
 
 
+def stratify_split(X, y, n_train, n_test, x_numba, y_numba, random_state):
+    """
+    Function to perform a stratified split based on y.
+    Identifies number of classes and samples per class, splices data within
+    each class.
+
+    Input:
+    X, y: shuffled input data and labels
+    """
+    if isinstance(X, cudf.DataFrame):
+        X = cp.fromDlpack(X.to_dlpack())
+    if isinstance(y, cudf.Series):
+        y = cp.fromDlpack(y.to_dlpack())
+    if isinstance(X, cuda.cudadrv.devicearray.DeviceNDArray):
+        X = cp.asarray(X)
+    if isinstance(y, cuda.cudadrv.devicearray.DeviceNDArray):
+        y = cp.asarray(y)
```

Comment:
just use the check `hasattr(object, "__cuda_array_interface__")` that is used afterwards, Numba arrays are cuda array interface compliant, no need for special treatment here